src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Workloads/WebMissingWorkloadDetector.cs

```diff
@@ -67,26 +62,25 @@ protected override async Task InitializeCoreAsync(CancellationToken cancellation
             // Note we don't use the ISafeProjectGuidService here because it is generally *not*
             // safe to use within IProjectDynamicLoadComponent.LoadAsync.
             _projectGuid = await _project.UnconfiguredProject.GetProjectGuidAsync();
-            _joinedDataSources = ProjectDataSources.JoinUpstreamDataSources(JoinableFactory, _projectFaultHandlerService, _projectSubscriptionService.ProjectSource, _wpfWorkloadDescriptorDataSource);
 
-            Action<IProjectVersionedValue<ValueTuple<IProjectSnapshot, ISet<WorkloadDescriptor>>>> action = OnWorkloadDescriptorsComputed;
+            Action<IProjectVersionedValue<ISet<WorkloadDescriptor>>> action = OnWorkloadDescriptorsComputed;
+
+            _subscription = new()
+            {
+                _webWorkloadDescriptorDataSource.SourceBlock.LinkTo(
+                    DataflowBlockFactory.CreateActionBlock(action, _project.UnconfiguredProject, ProjectFaultSeverity.LimitedFunctionality, nameFormat: $"{nameof(WebMissingWorkloadDetector)} Action: {1}"),
+                    linkOptions: DataflowOption.PropagateCompletion),
 
-            _subscription = ProjectDataSources.SyncLinkTo(
-                _projectSubscriptionService.ProjectSource.SourceBlock.SyncLinkOptions(),
-                _wpfWorkloadDescriptorDataSource.SourceBlock.SyncLinkOptions(),
-                    DataflowBlockFactory.CreateActionBlock(action, _project.UnconfiguredProject, ProjectFaultSeverity.LimitedFunctionality),
-                    linkOptions: DataflowOption.PropagateCompletion,
-                    cancellationToken: cancellationToken);
```

Comment:
This is probably the most impactful change here. It was sync-linking on `_projectSubscriptionService.ProjectSource` but not actually using that data source. `ProjectSource` fires very often, so it's good to avoid linking it. We also remove a synclink dataflow block in the process.